service/transactions/transactions.go

```diff
@@ -107,20 +118,25 @@ func (t *StorableTransaction) HandleResult(ctx context.Context, flowClient *clie
 	t.Error = ""
 
 	if result.Error != nil {
+		args, err := t.ArgumentsAsCadence()
+		if err != nil {
+			args = nil
+		}
+
+		logWithContext := log.WithFields(log.Fields{
+			"name":          t.Name,
+			"transactionID": t.TransactionID,
+			"error":         result.Error.Error(),
+			"arguments":     args,
+		})
+
 		t.Error = result.Error.Error()
 		if strings.Contains(result.Error.Error(), "invalid proposal key") {
```

Comment:
Does the error message contain an error code, `E1007` or smth? :) That would be safer to check for here. Could be updated in the same go even if this is not part of the PR.